From oliver@neukum.org  Mon Mar 15 16:15:34 2010
From: Oliver Neukum <oliver@neukum.org>
Date: Sat, 27 Feb 2010 20:57:12 +0100
Subject: usb: cdc-wdm: Fix deadlock between write and resume
To: Greg KH <greg@kroah.com>
Cc: Rickard Bellini <rickard.bellini@ericsson.com>, "linux-usb@vger.kernel.org" <linux-usb@vger.kernel.org>
Message-ID: <201002272057.12494.oliver@neukum.org>


From: Oliver Neukum <oliver@neukum.org>

The new runtime PM scheme allows resume() to have no locks.
This fixes the deadlock.

Signed-off-by: Oliver Neukum <neukum@b1-systems.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-wdm.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/class/cdc-wdm.c
+++ b/drivers/usb/class/cdc-wdm.c
@@ -839,10 +839,10 @@ static int wdm_resume(struct usb_interfa
 	int rv;
 
 	dev_dbg(&desc->intf->dev, "wdm%d_resume\n", intf->minor);
-	mutex_lock(&desc->lock);
+
 	clear_bit(WDM_SUSPENDING, &desc->flags);
 	rv = recover_from_urb_loss(desc);
-	mutex_unlock(&desc->lock);
+
 	return rv;
 }
 #endif
